src/Stream.ts

```diff
@@ -1,23 +1,25 @@
 import { UnsubscribeFn } from './ApiClient'
-import Message from './Message'
+import { MessageV1, MessageV2 } from './Message'
 import Client from './Client'
 import { messageApi, fetcher } from '@xmtp/proto'
 
-export type MessageTransformer<T> = (msg: Message) => T
+export type MessageDecoder<M> = (env: messageApi.Envelope) => Promise<M>
```

Comment:
MessageDecoder is to abstract out the V1 specific env->msg decoding.
Consequently the other types need to take message type parameter to support both versions.